Source/Core/RequestQueue.js

```diff
@@ -0,0 +1,186 @@
+define([
+        './Check',
+        './defineProperties'
+    ], function(
+        Check,
+        defineProperties) {
+    'use strict';
+
+    /**
+     * Priority queue for the {@link RequestScheduler} implemented as a sorted array.
+     * The request with the highest priority is placed at index 0 and the request
+     * with lowest priority is placed at index <code>length - 1</code>.
+     * <p>
+     * A lower <code>request.priority</code> value indicates that the request has higher priority. See {@link Request#priority}.
+     * </p>
+     *
+     * @alias RequestQueue
+     * @constructor
+     * @private
+     *
+     * @param {Number} maximumLength The maximum length of the queue.
+     */
+    function RequestQueue(maximumLength) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.number('maximumLength', maximumLength);
+        //>>includeEnd('debug');
+
+        this._array = new Array(maximumLength);
+        this._length = 0;
+        this._maximumLength = maximumLength;
```

Comment:
If this were public, there should be a getter for this.  As private, I guess it doesn't matter.